operator/internal/controller/redpanda/redpanda_controller_test.go

```diff
@@ -739,6 +738,10 @@ func (s *RedpandaControllerSuite) TestStableUIDAndGeneration() {
 		s.compareSnapshot(flipped, flippedBack, isStable)
 
 		s.deleteAndWait(rp)
+
+		var hr v2beta2.HelmRelease
```

Comment:
shouldn't this be checking for a `HelmChart`, not a `HelmRelease`?

For either resource, we'll want to include an assertion that the resource does exist at some point in time to ensure we're checking the right resource. I've seen too many tests like this pass because there's a typo 🙈

Reply:
I agree that I made mistake in the explicit check for HelmChart custom resource.

Never the less implicitly unique generated namespace could not be deleted if there are some resources hanging with finalizers. As I removed cleanup of left over HelmChart https://github.com/redpanda-data/redpanda-operator/pull/539/files#diff-0119f751e27a1f5c78bb3132cee1aee1b87644699eb02fab03e9a1c28edcb68fL835-L842 it proves that test pass and there is no HelmCharts left behind.

I will correct my assertion to be more explicit.